packages/components/tests/integration/components/hds/alert/index-test.js

```diff
@@ -0,0 +1,216 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, resetOnerror, setupOnerror } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Component | hds/alert/index', function (hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.afterEach(() => {
+    resetOnerror();
+  });
+
+  test('it renders the alert container', async function (assert) {
+    await render(hbs`<Hds::Alert @type="inline" @title="I'm a cool alert!" />`);
+    assert.dom(this.element).exists();
+  });
+  test('it should render with a CSS class that matches the component name', async function (assert) {
+    await render(
+      hbs`<Hds::Alert @type="inline" @title="I'm a cool alert!" id="test-alert" />`
+    );
+    assert.dom('#test-alert').hasClass('hds-alert');
+  });
+
+  // TYPE
+
+  test('it should render the correct CSS type class if @type prop is declared', async function (assert) {
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="yo" id="test-alert" />`
+    );
+    assert.dom('#test-alert').hasClass('hds-alert--type-inline');
+  });
+
+  // ICON
+
+  test('it should render an icon by default depending on the type and color', async function (assert) {
+    // here we don't test all the possible combinations, only some of them as precaution
+    assert.expect(6);
+    await render(hbs`<Hds::Alert @type="inline" @description="yo" />`);
+    assert.dom('.flight-icon-info').exists();
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="yo" @type="compact" />`
+    );
+    assert.dom('.flight-icon-info-fill').exists();
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="yo" @color="highlight" />`
+    );
+    assert.dom('.flight-icon-info').exists();
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="yo" @color="success" />`
+    );
+    assert.dom('.flight-icon-check-circle').exists();
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="yo" @color="warning" />`
+    );
+    assert.dom('.flight-icon-alert-triangle').exists();
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="yo" @color="critical" />`
+    );
+    assert.dom('.flight-icon-alert-diamond').exists();
+  });
+
+  test('if an icon is declared, the icon should render in the component and override the default one', async function (assert) {
```

Comment:
[suggestion] should there be a test here to ensure that the "Default one" renders?

Reply:
It's the previous test (`'it should render an icon by default depending on the type and color'`)